lib/cookie.c

```diff
@@ -819,7 +820,7 @@ Curl_cookie_add(struct Curl_easy *data,
      the same domain and path as this */
 
   /* at first, remove expired cookies */
-  remove_expired(c);
+  if(!noexpire) remove_expired(c);
```

Comment:
Nit: we put the conditional block on a new line below the if() line, never on the same!